### PR TITLE
New options that allow you to specify the buffer sizes.

### PR DESCRIPTION
**Pull request by roxlu**
This feature allows us to specify the buffer sizes that are used in Allocators.h and OpenGLDriver.cpp. Quick fix for #2057 By increasing these buffers we can render more instances.

### CODE REVIEW

File: filament/src/details/Allocators.h [hunk 1]
```diff
@@ -19,18 +19,33 @@
 
 #include <utils/Allocator.h>
 
+#ifndef FILAMENT_PER_RENDER_PASS_ARENA_SIZE_IN_MB
+#  define FILAMENT_PER_RENDER_PASS_ARENA_SIZE_IN_MB 2
+#error "wat"
```
Comment (romainguy) [inline filament/src/details/Allocators.h:24]: Did you mean to leave those #error?

File: filament/src/details/Allocators.h [hunk 2]
```diff
@@ -19,18 +19,33 @@
 
 #include <utils/Allocator.h>
 
+#ifndef FILAMENT_PER_RENDER_PASS_ARENA_SIZE_IN_MB
+#  define FILAMENT_PER_RENDER_PASS_ARENA_SIZE_IN_MB 2
```
Comment (romainguy) [inline filament/src/details/Allocators.h:23]: Our indentation style is 4 spaces, not 2

File: CMakeLists.txt [hunk 3]
```diff
@@ -21,13 +21,32 @@ option(FILAMENT_ENABLE_LTO "Enable link-time optimizations if supported by the c
...
 # ==================================================================================================
 if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.12")
     cmake_policy(SET CMP0074 NEW)
 endif()
 
+# ==================================================================================================
+# Defines
+# ==================================================================================================
+
+add_definitions(
```
Comment (romainguy) [inline CMakeLists.txt:43]: This should go in filament/CMakeLists.txt, not the root one. There's a `Compiler flags` section there.

File: filament/src/details/Allocators.h (hunk 1 above)
Comment (roxlu) [reply filament/src/details/Allocators.h]: No :) removing them.

File: CMakeLists.txt (hunk 3 above)
Comment (roxlu) [reply CMakeLists.txt:43]: Do you want to keep the options in the main one and de `add_definitions()` in `filament/CMakeLists.txt` ?

File: filament/src/details/Allocators.h (hunk 2 above)
Comment (roxlu) [reply filament/src/details/Allocators.h:23]: Will fix.

File: CMakeLists.txt [hunk 4]
```diff
@@ -21,6 +21,14 @@ option(FILAMENT_ENABLE_LTO "Enable link-time optimizations if supported by the c
 
 option(FILAMENT_SKIP_SAMPLES "Don't build samples" OFF)
 
+set(FILAMENT_PER_RENDER_PASS_ARENA_SIZE_IN_MB "2" CACHE STRING "Per render pass arena size. Must be roughly 1 MB larger than FILAMENT_PER_FRAME_COMMANDS_SIZE_IN_MB, default 2.")
+
+set(FILAMENT_PER_FRAME_COMMANDS_SIZE_IN_MB "1" CACHE STRING "Size of the high-level draw commands buffer. Rule of thumb, 1 MB less than FILAMENT_PER_RENDER_PASS_ARENA_SIZE_IN_MB, default 1.")
+
+set(FILAMENT_MIN_COMMAND_BUFFERS_SIZE_IN_MB "1" CACHE STRING "Size of the command-stream buffer. As a rule of thumb use the same value as FILAMENT_PER_FRRAME_COMMANDS_SIZE_IN_MB, default 1.")
```
Comment (bejado) [inline CMakeLists.txt:28]: FRRAME -> FRAME

File: filament/backend/src/opengl/OpenGLDriver.cpp [hunk 5]
```diff
@@ -137,10 +137,10 @@ OpenGLDriver::DebugMarker::~DebugMarker() noexcept {
 
 OpenGLDriver::OpenGLDriver(OpenGLPlatform* platform) noexcept
         : DriverBase(new ConcreteDispatcher<OpenGLDriver>()),
-          mHandleArena("Handles", 2U * 1024U * 1024U), // TODO: set the amount in configuration
+          mHandleArena("Handles", FILAMENT_OPENGL_HANDLE_ARENA_SIZE_IN_MB * 1024U * 1024U), // TODO: set the amount in configuration
```
Comment (bejado) [inline filament/backend/src/opengl/OpenGLDriver.cpp:140]: We can probably remove that TODO comment now

Comment (pixelflinger) [reply filament/backend/src/opengl/OpenGLDriver.cpp:140]: @bejado I would like to make those a runtime config option when you create the engine, someday.

File: CMakeLists.txt (hunk 4 above)
Comment (roxlu) [reply CMakeLists.txt:28]: I guess it’s time to get some sleep first :-)

File: filament/backend/src/opengl/OpenGLDriver.cpp (hunk 5 above)
Comment (BStringhamVRSK) [reply filament/backend/src/opengl/OpenGLDriver.cpp:140]: I am looking forward to having this change in the next release.  Our scenes are also potentially large and complex.
I *really* look forward to having runtime configurable options!

Comment (romainguy) [reply filament/backend/src/opengl/OpenGLDriver.cpp:140]: Note that this is not runtime configurable with this change.

**Pull request merged**